tests/test_process.cc

```diff
@@ -0,0 +1,77 @@
+#include "gtest/gtest.h"
+
+#include "process.h"
+
+using process = h5s3::testing::process;
+
+
+TEST(process, clean_exit) {
+    std::vector<std::string> args = {"python", "-c", "exit(0)"};
+    process::environment env = {};
+    process proc(args, env);
+
+    auto [code, state] = proc.join();
+    EXPECT_EQ(code, 0);
+    EXPECT_EQ(state, process::state::EXITED);
+}
+
+TEST(process, error_exit) {
+    std::vector<std::string> args = {"python", "-c", "exit(1)"};
+    process::environment env = {};
+    process proc(args, env);
+
+    auto [code, state] = proc.join();
+    EXPECT_EQ(code, 1);
+    EXPECT_EQ(state, process::state::EXITED);
+}
+
+TEST(process, environment) {
+    std::vector<std::string> args = {"python", "-c", "import os; assert os.environ['foo'] == 'bar'"};
+    process::environment env = {{"foo", "bar"}};
+    process proc(args, env);
+
+    auto [code, state] = proc.join();
+    EXPECT_EQ(code, 0);
+    EXPECT_EQ(state, process::state::EXITED);
+}
+
+TEST(process, interrupt) {
+    std::vector<std::string> args = {"python", "-c", "while True:\n    pass"};
```

Comment:
If this doesn't actually work, won't we be stuck forever? What if we just ran something like `sleep` so that we can at least continue execution.